### PR TITLE
Create @next AttributeFilter section

### PR DESCRIPTION
**Pull request by ivanmjartan**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeDropdown/types.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// (C) 2019-2022 GoodData Corporation
+import { IPagedResource } from "@gooddata/sdk-backend-spi";
+import { IAttributeElement } from "@gooddata/sdk-model";
+
+export const emptyListItem: EmptyListItem = { empty: true };
+
+/**
+ * @public
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeDropdown/types.ts:8]: `@internal` (same for other interfaces in this file)

File: libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    attributeElementsToAttributeElementArray,
+    getObjRef,
+    getParentFilterTitles,
+    getValidElementsFilters,
+} from "./utils/AttributeFilterUtils";
+import invariant from "ts-invariant";
+import stringify from "json-stable-stringify";
+
+/**
+ * @public
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeFilter.tsx:42]: again, please mark everything as `@internal`, no public annotation should be in the `AttributeFilter@next` folder for now.

**Review by kandl — CHANGES_REQUESTED** [review]
Please remove all `@public` annotations for now, also it seems you have misspelled the folder name `AttributeFIlter@next` (should be `AttributeFilter@next`)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8089/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9905/ : FAILURE in 11m 39s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeDropdown/types.ts (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeDropdown/types.ts]: fixed

File: libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeFilter.tsx (hunk 2 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFIlter@next/AttributeFilter.tsx]: fixed

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8092/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9908/ : FAILURE in 14m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8093/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9909/ : SUCCESS in 12m 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8096/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9911/ : SUCCESS in 12m 16s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8108/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9924/ : FAILURE in 8m 42s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8109/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9925/ : SUCCESS in 12m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8112/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9928/ : SUCCESS in 12m 45s

File: libs/sdk-ui-filters/src/index.ts [hunk 3]
```diff
@@ -76,3 +76,25 @@ export {
     IAttributeDropdownItem,
     ICustomGranularitySelection,
 } from "./RankingFilter/types";
+
```
Comment (kandl) [inline libs/sdk-ui-filters/src/index.ts:79]: Please remove the comented code :)

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2551

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2313/ : SUCCESS in 2m 18s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2605/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1425/ : SUCCESS in 19m 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9931/ : SUCCESS in 12m 51s

**Pull request merged**